### ISSUE TITLE
Backspace key moves default block into Group block instead of deleting the default block

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#65850]
### Description

When pressing the Backspace key (del) in the default block or in an empty paragraph, I expect the default block to be deleted and the previous block to be selected.

This works in most cases, but it doesn't work when the previous block is a Group. Note: I haven't tested with other blocks so it would be great to double check whether this inconsistent behavior happens only with the Group block os also with other ones. As an user, I'm totally confused by this inconsistent behavior. I learned that backspace is supposed to delete the empty default block but in some cases it does something different.

It is worth mentioning that at first I thought the Group block was being deleted instead of the default block. Turns out the Group block isn't deleted. Instead the default block gets moved into the Group block thus becoming a nested paragraph. As an user, this behavior is totally unexpected to me and breaks an established interaction pattern where I expect the backspace key to just delete the empty paragraph.

I think providing a consistent, predictable behavior is key to provide a cohesive user experience. The behavior of the backspace key should be standardized ans always perform the same action. Adding tests would be beneficial as well.

Animated GIF to illustrate:

![Image](https://github.com/user-attachments/assets/a46d4cd0-31fe-426c-9a53-9b37b2630d6d)

### Step-by-step reproduction instructions

- Edit a post.
- Open the List View to better check what happens with the blocks deletion and selection.
- Add a Table block.
- Click the empty area below the table block to add the default paragraph block.
- Click again to select the default paragraph block and focus it.
- Press the backspace key.
- Observe the default paragraph block gets deleted and the Table block gets selected.
- Add a Columns block.
- Repeat the steps above and observe the behavior is consistent.
- Add a Group block.
- Repeat the steps above.
- Observe that, visually in the canvas, it looks like the Group block gets deleted and the default block stays, which is extremely confusing.
- By observing the List view you can get what actually happens.
- Observe the default paragraph block isn't deleted and instead it gets moved into the Group block.
- Expected: the default paragraph block to be deleted and the Group block to be selected.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by Mamaduka** [comment]
@kevin940726, since you worked on [recent fixes](https://github.com/WordPress/gutenberg/pull/65262) in the area; is this an expected behavior?

**Comment by kevin940726** [comment]
Yeah I think this behavior is unrelated to the recent fix and has been here for a while now. I'm not sure if this is an expected behavior, though. I agree it would be nice to handle this, even treat it as a special case if necessary.

**Comment by afercia** [comment]
> Yeah I think this behavior is unrelated to the recent fix and has been here for a while now.

I forgot to mention this can be reproduced in WordPress 6.6 and 6.5 as well (haven't tested earlier releases) so yes it's been around for a while. Still, it's an inconsistent and unexpected beheavior. I think the backspace key should always do the same thing.

**Comment by megane9988** [comment]
https://github.com/user-attachments/assets/b01dfcdc-b15c-4dbb-b90b-357aaebde63f

+1


The heading blocks following image blocks cannot be deleted with just the backspace key either.

- WordPress 6.8.1 
- Gutenberg Version 21.1.0

**Comment by t-hamano** [comment]
> It is worth mentioning that at first I thought the Group block was being deleted instead of the default block. Turns out the Group block isn't deleted. Instead the default block gets moved into the Group block thus becoming a nested paragraph.

This is an expected behaviour. See https://github.com/WordPress/gutenberg/pull/53508

In other words, because [the Group block supports `__experimentalOnMerge`](https://github.com/WordPress/gutenberg/blob/5d2f86a1e0db8708d875b1ca04f9e25541a822b0/packages/block-library/src/group/block.json#L25), the Group block has special behavior when blocks are merged.

I suspect one of the reasons this behavior is confusing is that even though the Group block doesn't have any child blocks, i.e. the placeholder is displayed, when the blocks are merged, the placeholder suddenly disappears.

I don't think this is a bug, but rather a specification issue.

Two approaches come to mind:

- **Delete** `__experimentalOnMerge` support of the Group block. This means that when the backspace key is pressed in a block next to a Group block, the focus will always move to the Group block.
- **Keep** `__experimentalOnMerge` support of the Group block. However, when a container block with `__experimentalOnMerge` hook has no child blocks, we may want to disable nested merging and focus the container block.

> The heading blocks following image blocks cannot be deleted with just the backspace key either.

This is probably a separate issue from this one, so you might want to submit a new one.

As I recall, when merging blocks, there was special processing that only worked for the default block (i.e. the Paragraph block).
